### ISSUE TITLE
Allow user to specify default CRI-O alias and default to docker.io

### ISSUE THREAD

**Issue by jefflill** [nforgeio/neonKUBE#1310]
I'm starting to play around with the [Bridge to Kubernetes Sample](https://docs.microsoft.com/en-us/visualstudio/bridge/bridge-to-kubernetes-vs?view=vs-2019&WT.mc_id=visualstudio_bridge_aka_bridge-to-k8s-vs-quickstart) tutorial in a neonKUBE cluster and after applying the manifests, the pods aren't starting due to **ImageInspectError**.

After poking around a bit, I realized that the problem is that the pod container images are hosted on **docker.io** and these are referenced like **azdspublic/bikesharing-reservationengine** instead of **docker.io/azdspublic/bikesharing-reservationengine**.  Other samples use Helm charts to deploy which means image repository references will be all over the place,

The problem here is that neonDESKTOP is intended for beginning Kubernetes users and many of them will go looking for tutorials about doing Kubernetes stuff on the web and it's very likely that many or most of these tutorials won't work out-of-the-box because they assume **docker.io** is the default registry and I assume that this will likely be the case for most other Kubernetes distributions because docker has been the default CRI for so long.

@marcusbooyah: I know you had configured CRI-O with default aliases several months ago and then we decided to drop that for some reason.  I don't remember exactly why, but it may have been due to the chaos Docker caused when they start imposing rate limits a year ago.

I think we'll probably need to address this, we don't want many user's first experience with neonKUBE to be disappointing and then require them to modify Kubernetes manifests and Helm charts right before anything will work and before they have much of a clue about what those things even are.

* Should we set the CRI-O default image registry to **docker.io**?
* Should we allow this default to be specified in the cluster definition?
* Should we allow the default to be changed after cluster creation by updating the cluster definition CR and having an operator update the nodes?
* We'd also probably need a way for users to specify their docker or other registry credentials, right?  This seems like a required feature anyway so that users can have private image registries.

**Comment by jefflill** [comment]
Here's how this is landing:

* I've added the new **ClusterDefinition.DefaultContainerRegistry** property.  This may be set to the registry's DNS host name or IP address and defaults to **docker.io**.  Setting this to NULL or an empty string disables default registry lookups.

* I had to remove the **idempotent** check in the **NodeSshProxy.NodeInstallCriO()** method because we'll need to re-run this during cluster setup, otherwise we'd never override what was configured for the node images.

* I also added a call to **NodeSshProxy.NodeInstallCriO()** in cluster setup so that the CRI-O will be reconfigured, but I added a parameter so that we won't re-install CRI-O.
